tests/test_radius_server.rb

```diff
@@ -0,0 +1,182 @@
+#
+# Minitest for RadiusServer class
+#
+# Copyright (c) 2014-2015 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require File.expand_path('../ciscotest', __FILE__)
+require File.expand_path('../../lib/cisco_node_utils/radius_server', __FILE__)
+
+# TestRadiusServer - Minitest for RadiusServer node utility.
+class TestRadiusServer < CiscoTestCase
+  def setup
+    # setup runs at the beginning of each test
+    super
+    no_radiusserver
+  end
+
+  def teardown
+    # teardown runs at the end of each test
+    no_radiusserver
+    super
+  end
+
+  def no_radiusserver
+    # Turn the feature off for a clean test.
+    config('no logging server 8.8.8.8',
+           'no logging server 9.9.9.9')
+  end
+
+  # TESTS
+
+  def test_radiusserver_create_destroy_single # rubocop:disable Metrics/AbcSize
+    id = '8.8.8.8'
+    refute_includes(Cisco::RadiusServer.radiusservers, id)
+
+    server = Cisco::RadiusServer.new(id, true)
+    assert_includes(Cisco::RadiusServer.radiusservers, id)
+    assert_equal(Cisco::RadiusServer.radiusservers[id], server)
+
+    server.accounting_only = true
+    assert_equal(Cisco::RadiusServer.radiusservers[id].accounting_only,
+                 true)
+    assert_equal(server.accounting_only,
+                 true)
+
+    server.authentication_only = true
+    assert_equal(Cisco::RadiusServer.radiusservers[id].authentication_only,
+                 true)
+    assert_equal(server.authentication_only,
+                 true)
+
+    server.acct_port = 44
+    assert_equal(Cisco::RadiusServer.radiusservers[id].acct_port,
+                 44)
+    assert_equal(server.acct_port,
+                 44)
```

Comment:
Not sure I see the benefit of checking two different instances referring to the same configuration.